File: ios/GalaxyRN/GxyUIState/GxyUIStateModule.m
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(GxyUIStateModule, RCTEventEmitter)

RCT_EXTERN_METHOD(startForeground)
RCT_EXTERN_METHOD(stopForeground)

@end 

File: ios/GalaxyRN/GxyUIState/GxyUIStateModule.swift
import AVFoundation
import CallKit
import Foundation
import React
import UIKit

@objc(GxyUIStateModule)
class GxyUIStateModule: RCTEventEmitter {
    // MARK: - Properties

    // MARK: - Initialization
    override init() {
        super.init()
    }

    // MARK: - Public Methods

    @objc(startForeground)
    func startForeground() {
        DispatchQueue.main.async {
            UIApplication.shared.isIdleTimerDisabled = true
        }
    }

    @objc(stopForeground)
    func stopForeground() {
        DispatchQueue.main.async {
            UIApplication.shared.isIdleTimerDisabled = false
        }
    }

    @objc
    override static func moduleName() -> String! {
        return "GxyUIStateModule"
    }
}
